pandas/tests/test_base.py

```diff
@@ -1365,8 +1365,8 @@ def mock_import_fail(name, *args, **kwargs):
         "\ndateutil: cannot import name some_other_dependency"
     )
 
-    with patch("builtins.__import__") as mock_import:
-        mock_import.side_effect = mock_import_fail
+    with monkeypatch.context() as m:
```

Comment:
this test should be elsewhere


maybe test_downstream

Reply:
That looks like a better place, thanks.

Reply:
This has nothing to do with optional downstream packages though ... so I don't think it fits here either.

Reply:
Does `test_downstream` explicitly handle only optional dependencies? The docstring in that file isn't that specific - it simply refers to "downstream packages".

Reply:
sorry, "optional" was misleading a bit. But in any case: all the tests here are for packages that depend on pandas (downstream packages, so are by definition not required packages for pandas). 
Here we want to test the import error message *in pandas* for the few required hard dependencies we have, which thus has not much to do with testing downstream packages.

Reply:
Ah, that makes more sense! I had been thinking that upstream/downstream seemed different to the normal way round.